homeassistant/components/hyperion/light.py

```diff
@@ -78,34 +110,194 @@
+        if entity.config_entry_id is not None or entity.platform != DOMAIN:
+            continue
+        result = re.search(rf"([^:]+):(\d+)-{instance}", entity.unique_id)
+        if result and result.group(1) == host and int(result.group(2)) == port:
+            registry.async_update_entity(entity_id, new_unique_id=future_unique_id)
+            break
+    else:
+        # Possibility 3: This is the first upgrade to the new Hyperion component.
+        # No config entry and no entity_registry entry, in which case the CONF_NAME
+        # variable will be used as the preferred name. Rather than pollute the config
+        # entry with a "suggested name" type variable, instead create an entry in the
+        # registry that will subsequently be used when the entity is created with this
+        # unique_id.
+
+        # This also covers the case that should not occur in the wild (no config entry,
+        # but new style unique_id).
+        registry.async_get_or_create(
+            domain=LIGHT_DOMAIN,
+            platform=DOMAIN,
+            unique_id=future_unique_id,
+            suggested_object_id=config[CONF_NAME],
+        )
+
+    async def migrate_yaml_to_config_entry_and_options(host, port, priority):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_HOST: host,
+                CONF_PORT: port,
+            },
+        )
+        if (
+            result["type"] != data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+            or result.get("result") is None
+        ):
+            _LOGGER.warning(
+                "Could not automatically migrate Hyperion YAML to a config entry."
+            )
+            return
+        # At a later stage add options migration functionality.
+        _LOGGER.info(
+            "Successfully migrated Hyperion YAML configuration to a config entry."
+        )
+
+    # Kick off a config flow to create the config entry.
+    hass.async_create_task(
+        migrate_yaml_to_config_entry_and_options(host, port, config[CONF_PRIORITY])
+    )
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up a Hyperion platform from config entry."""
+    host = config_entry.data[CONF_HOST]
+    port = config_entry.data[CONF_PORT]
+    token = config_entry.data.get(CONF_TOKEN)
+
+    async def async_instances_to_entities(response: Dict[str, Any]) -> None:
+        if not response or const.KEY_DATA not in response:
+            return
+        await async_instances_to_entities_raw(response[const.KEY_DATA])
+
+    async def async_instances_to_entities_raw(instances: Dict[str, Any]) -> None:
+        registry = await async_get_registry(hass)
+        entities_to_add: List[Hyperion] = []
+        desired_unique_ids: Set[str] = set()
+        current_entities = hass.data[DOMAIN][config_entry.entry_id][LIGHT_DOMAIN][
```

Comment:
We don't need to define this variable again here. We can access the variable that is available in the outer scope.

Reply:
@MartinHjelmare -- can you clarify, which variable are you referring to? If it's `current_entities`, then this is really just a convenience to not need to use hass.data[...][...][...][...], and I don't see this readily available at the outer scope?

Reply:
Ah, I think you're referring to `current_entities` which _could be_ at the outer scope, as per your suggestion below.